content/technotes/2020-02-26-tic-ghactions.Rmd

```diff
@@ -0,0 +1,170 @@
+
+#### macOS compiler toolchain
+
+macOS is a complicated platform for R.
+CRAN does use a custom compiler for creating the binaries for this platform.
+One reason is that the default `/usr/bin/clang` does not support `openMP`.
+For the current R 3.6.x release a custom version of `clang7` is used while for R 4.x `clang8` will be used.
+tic reflects exactly this setting on the macOS-release and macOS-devel runners, eliminating lots of issues when compiling packages from source and giving you the security to have the CRAN setup mirrored in your builds.
+
+#### Addtional tweaks
+
+- tic always runs `R CMD javareconf` to ensure a working installation of rJava.
+- tic uses four cores for installing dependencies instead of just running sequentially.
+- tic sets the env var `RGL_USE_NULL` to make it possible to install the rgl package if needed.
+- tic installs all packages from source on Linux instead of using the R package binaries provided by RStudio.
+  This is to mimic the normal behavior on Linux and ensure a proper linking against system libraries.
+    
+### Getting Started
+
+If you are triggered now to try out tic and GitHub Actions, here is how to get started (alternatively you can also run `tic::use_tic()` to start an interactive wizard guiding you through the setup process):
+
+```{r, eval=FALSE}
+library("tic")
+
+use_ghactions_deploy() # (optional) setup deployment from builds
+use_ghactions_yml() # use deploy = TRUE to deploy from builds
+tic::use_tic_r(repo_type = "package", deploy_on = "ghactions") # add tic.R
+use_tic_badge("ghactions") # (optional) add a badge to README.md/README.Rmd
+```
+
+After that, commit `.github/workflows/main.yml` and `tic.R` and have a look at the "Actions" pane in the repository.
+For a full example of the resulting YAML file you can have a look at our example repository for packages [tic.package](https://github.com/ropenscilabs/tic.package/blob/25fe68d3d874a671e8a1350dfd47789d7dd19105/.github/workflows/main.yml).
+
+### Summary
+
+tic aims to simplify CI related tasks for R projects (not just packages by the way) and now provides supercharged support for GitHub Actions.
+By enhancing the good work of [r-lib/actions](https://github.com/r-lib/actions) tic provides a DSL for R for CI tasks.
+
+If you encounter problems, please have a look at the vignettes of tic, especially the ["FAQ"](https://docs.ropensci.org/tic/articles/faq.html) one.
+You can also open an issue in the repo if something does not work as planned.
+tic will keep improving the existing templates and its internal functionality[^1].
+
+We have plans to add an update mechanism in the future to simplify staying up-to-date with the latest template changes upstream - stay tuned!
+
+[^1]: Since tic is not on CRAN yet, you have to watch the development on GitHub for now to stay up-to-date.
```

Comment:
Consider adding an Acknowledgements section with links to the review repository for tic and reviewers author pages.

Reply:
I haven't done so because it's a technote and not a package announcement - should I still do it?

Reply:
Yes please :grin: It's always nice to showcase the review process and the reviewers, especially since this is the first post about tic on rOpenSci. In addition, I always find my confidence in a package goes up if I know that it has been officially reviewed and vetted by other people not part of the project.